src/ui/public/notify/__tests__/notifier.js

```diff
@@ -43,8 +51,47 @@ describe('Notifier', function () {
       expect(notify('error').title).to.equal('Error');
     });
 
-    it('sets lifetime to Infinity', function () {
-      expect(notify('error').lifetime).to.equal(Infinity);
+    it('sets lifetime to 5 minutes', function () {
+      expect(notify('error').lifetime).to.equal(300000);
+    });
+
+    it('sets timeRemaining and deprecates', function () {
+      var notif = notify('error');
+
+      expect(notif.timeRemaining).to.equal(300);
+      $interval.flush(1000);
```

Comment:
Looking over the implementation for $interval.flush, it appears to be guaranteed. I have also not seen a single failure due to timing.

https://github.com/angular/angular.js/blob/master/src/ngMock/angular-mocks.js#L527